lib/barbeque/message_handler/job_retry.rb

```diff
@@ -16,8 +16,11 @@ def initialize(message:, job_queue:)
       end
 
       def run
-        job_retry = Barbeque::JobRetry.find_or_initialize_by(message_id: @message.id)
-        job_retry.update!(job_execution: job_execution)
+        begin
+          job_retry = Barbeque::JobRetry.create(message_id: @message.id, job_execution: job_execution)
```

Comment:
And, only with deduping both JobExecution and JobRetry, we can assume that the same message isn't consumed by multiple workers. People who write code of job won't be necessary to care about exclusive control.